controller/api/public/client.go

```diff
@@ -190,6 +190,7 @@ func newClient(apiURL *url.URL, httpClientToUse *http.Client, controlPlaneNamesp
 	}, nil
 }
 
+// NewInternalClient creates a new Public API client intended to run inside a Kubernetes cluster.
```

Comment:
Mind wrapping this at 80 characters? Same goes for the comment on line 203 in this file.